requirements.txt

```
@@ -31,11 +31,10 @@ rouge-score>=0.1.2,<0.2.0
 sentencepiece>=0.2.0,<0.3.0
 tokenizers>=0.15.2,<0.16.0
 toml>=0.10.2,<0.11.0
-# Habana installer has 2.2.0a0+git8964477 with oneMKL
 torch>=2.2.0a0,<3.0.0 ; python_version == '3.10'
-torch>=2.2.1,<3.0.0 ; python_version != '3.10'
+torch>=2.3.0,<3.0.0 ; python_version != '3.10'
 tqdm>=4.66.2,<5.0.0
-transformers>=4.30.0,<=4.38.2
+transformers==4.38.1
```

The optional extra dependencies are a clean, maintainable solution. A quick hack is a platform-specific version pin. It abuses the fact that MPS is only available on macOS ARM64:

```suggestion
transformers==4.38.1; sys_platform == 'darwin' and platform_machine == 'arm64'
transformers>=4.37.0,<=4.38.2; sys_platform != 'darwin' and platform_machine != 'arm64'
```

I'll try to come up with a better solution next week.

---

oh this is great. Thanks!